frame/balances/src/lib.rs

```diff
@@ -1295,6 +1317,194 @@ impl<T: Config<I>, I: Instance> ReservableCurrency<T::AccountId> for Module<T, I
 	}
 }
 
+impl<T: Config<I>, I: Instance> NamedReservableCurrency<T::AccountId> for Module<T, I>  where
+	T::Balance: MaybeSerializeDeserialize + Debug
+{
+	fn reserved_balance_named(id: &ReserveIdentifier, who: &T::AccountId) -> Self::Balance {
+		let reserves = Self::reserves(who);
```

Comment:
Note that despite the binary search, this is still `O(named_reservations)` (due to needing the full deserialisation of `Self::reserves`).